src/sync/mod.rs

```diff
@@ -9,3 +9,4 @@ mod atomic_option;
 mod ms_queue;
 mod treiber_stack;
 mod seg_queue;
+pub mod deque;
```

Comment:
If the only exported type from this module were `Deque` I'd want to just reexport it, but with a `deque` constructor, `Worker`, `Stealer`, and `Steal` types I figured it may be best for now to just expose the module.

Open to suggestions though!